addons/mrp/models/stock_rule.py

```diff
@@ -118,9 +118,13 @@ def _prepare_mo_vals(self, product_id, product_qty, product_uom, location_id, na
         }
         if location_id.get_warehouse().manufacture_steps == 'pbm_sam':
             # Use the procurement group created in _run_pull mrp override
+            # Preserve the origin from the original stock move, if available
+            if len(values.get('move_dest_ids', [])) == 1 and values['move_dest_ids'][0].origin:
```

Comment:
Hum my bad, a few lines above we do:

```py
'move_dest_ids': values.get('move_dest_ids') and [(4, x.id) for x in values['move_dest_ids']] or False,
```

So it must be a recordset since we access `x.id`. `values['move_dest_ids'].origin` is fine.

I'll check this tomorrow in more details, though.

Reply:
In theory you could have something like 
```
'move_dest_ids': [ my_move ],
```